CONTRIBUTING.md

```diff
@@ -1 +1,60 @@
-See the [Flynn contributing guide](https://flynn.io/docs/contributing).
+# Contributing Guide
+
+We welcome and encourage community contributions to go-tuf.
+
+Please familiarize yourself with the Contribution Guidelines before contributing.
+
+There are many ways to help go-tuf besides contributing code:
+  - Fix bugs or file issues
+  - Provide feedback on the CLI experience or suggest feature enhancements.
+  - Improve documentation.
+
+## Contributing Code
+
+Unless you are fixing a known bug, we strongly recommend discussing it with the community via a GitHub issue or Slack before getting started to ensure that your work is consistent with TUF's specification.
+
+All contributions are made via pull request. All patches from all contributors get reviewed. See the Pull Request procedure.
+
+
+## Pull Request Procedure
+
+To make a pull request, you will need a GitHub account. See GitHub's documentation [forking](https://help.github.com/articles/fork-a-repo) and [pull requests](https://help.github.com/articles/using-pull-requests). 
+
+Pull requests should be targeted at the `master` branch. Before creating a pull request, go through this checklist:
+
+1. Create a feature branch off of `master` so that changes do not get mixed up.
+2. If your PR adds new code, it should include tests covering the new code. If your PR fixes a bug, it should include a regression test.
+3. PRs that change user-facing behavior or CLI must have associated documentation.
+4. All code comments and documentation are expected to have proper English grammar and punctuation. 
+5. [Rebase](http://git-scm.com/book/en/Git-Branching-Rebasing) your local changes against the `master` branch.
+6. Run the full project test suite with the `go test ./...` command and confirm that it passes.
+7. Run `go fmt ./...`.
+
+When creating a PR, 
+
+1. Accept the Developer's Certificate of Origin on all commits (see above).
```

Comment:
If we require this, probably worth adding the [DCO bot](https://probot.github.io/apps/dco/)

Reply:
I _hate_ that DCO bot. Can we instead use an app that checks whether a contributor has signed a DCO agreement? Something like what the Linux Foundation uses.

Reply:
:rofl: :rofl:  I'm not attached to any particular bot, just saying that if it's "required" that should be enforced :)

Though I can't find what you're talking about -- I checked a few LF repos and the ones I found all use the same bot ([ex.](https://github.com/LF-Engineering/kelpy/pull/4/checks)).

Reply:
Sorry, I meant a CLA bot, which many orgs have their own apps for, here's the LF one: https://github.com/apps/lf-cla-application

Reply:
TLDR: no to DCO, yes to CLA

Reply:
Why the preference for CLA? 

As I understand it (I am not a lawyer) both are assertions that the submitter has the right to submit a change and grants the right to the receiver to use the submitted change. A CLA is a legal contract while the DCO is a more lightweight statement of the contributor asserting ownership. As a consequence of a CLA being a legal contract, they can be a barrier to contributors (requiring, i.e., a corporate legal representative to sign-off).

Reply:
Simply because DCO is extremely annoying (must remember to `--sign-off`, otherwise rebase and force-push hell; does not work with commits done from the website), whereas CLA is a one-time deal.